ecs-init/gpu/nvidia_gpu_manager.go

```diff
@@ -65,6 +69,8 @@ func NewNvidiaGPUManager() GPUManager {
 
 // Setup is used for setting up gpu information in the instance
 func (n *NvidiaGPUManager) Setup() error {
```

Comment:
When is this function called? (i.e. under what conditions?): 
  - If GPUs are present on the box?
  - If ECS is configured to use GPUs?
  - If the drivers are installed?

Reply:
If `ECS_ENABLE_GPU_SUPPORT` is set to true, this func will be called (during pre-start process).